scripts/dts/extract/clocks.py

```diff
@@ -0,0 +1,150 @@
+#
+# Copyright (c) 2018 Bobby Noelte
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+from extract.globals import *
+from extract.directive import DTDirective
+
+##
+# @brief Manage clocks related directives.
+#
+# Handles:
+# - clocks
+# directives.
+#
+class DTClocks(DTDirective):
+
+    def __init__(self):
+        pass
+
+    def _extract_consumer(self, node_address, yaml, clocks, names, defs, def_label):
```

Comment:
Can you remove this module in the PR?
This is new material that would deserve a dedicated review with matching yaml files and examples.

Reply:
This is the clocks directive (eg. BUS, BITS) as before.

Reply:
Yeah, and this is actually a bit more than that as you added #clock-cells treatment.
I don't say this is not correct, but that this is more than the initial status of script features.

I've checked what was generated  and it seems true to what was generated earlier.
What puzzles me is that there is more code than in initial function that was generic to other nodes..